vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/credentialverifier/validator/LdpValidator.kt

```diff
@@ -0,0 +1,167 @@
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.ISSUANCE_DATE
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.ISSUER
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.PROOF
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.REFRESH_SERVICE
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.TERMS_OF_USE
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.TYPE
+import io.mosip.vercred.vcverifier.constants.CredentialValidatorConstants.VALID_UNTIL
+import io.mosip.vercred.vcverifier.data.DATA_MODEL
+import io.mosip.vercred.vcverifier.data.VerificationResult
+import io.mosip.vercred.vcverifier.exception.ValidationException
+import io.mosip.vercred.vcverifier.utils.DateUtils
+import io.mosip.vercred.vcverifier.utils.Util
+import io.mosip.vercred.vcverifier.utils.ValidationHelper
+import org.bitcoinj.core.VerificationException
+import org.json.JSONObject
+
+class LdpValidator {
+
+    private val commonMandatoryFields = listOf(
+        CONTEXT,
+        TYPE,
+        CREDENTIAL_SUBJECT,
+        ISSUER,
+        PROOF
+    )
+
+    private val commonIDMandatoryFields = listOf(
+        CREDENTIAL_SCHEMA
+    )
+
+    //All Fields has Type Property as mandatory with few fields as ID as optional.
+    private val fieldsWithIDAndType = listOf(
+        PROOF,
+        CREDENTIAL_STATUS,
+        EVIDENCE,
+        CREDENTIAL_SCHEMA,
+        REFRESH_SERVICE,
+        TERMS_OF_USE
+    )
+
+    private val validationHelper = ValidationHelper()
+    private val dateUtils = DateUtils()
+
+    fun validate(credential: String): VerificationResult {
+        try {
+            if (credential.isEmpty()) {
+                throw ValidationException(ERROR_EMPTY_VC_JSON)
+            }
+
+            val vcJsonObject = JSONObject(credential)
+
+            val contextVersion = Util().getContextVersion(vcJsonObject)
+                ?: throw ValidationException("$ERROR_MISSING_REQUIRED_FIELDS$CONTEXT")
+            when (contextVersion) {
+                DATA_MODEL.DATA_MODEL_1_1 -> {
+                    validateV1SpecificFields(vcJsonObject)
+                    validateCommonFields(vcJsonObject)
+                    val expirationMessage = if (vcJsonObject.has(EXPIRATION_DATE) && dateUtils.isVCExpired(vcJsonObject.optString(
+                            EXPIRATION_DATE))) ERROR_VC_EXPIRED else ""
+                    return VerificationResult(true, expirationMessage)
+                }
+                DATA_MODEL.DATA_MODEL_2_0 -> {
+                    validateV2SpecificFields(vcJsonObject)
+                    validateCommonFields(vcJsonObject)
+                    val expirationMessage = if (vcJsonObject.has(VALID_UNTIL) && dateUtils.isVCExpired(vcJsonObject.optString(VALID_UNTIL))) ERROR_VC_EXPIRED else ""
+                    return VerificationResult(true, expirationMessage)
```

Comment:
return only error message as empty or actual error.

Reply:
Updated to return ErrorMessage